src/OrbitQt/TrackTypeItemModel.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_TRACK_TYPE_ITEM_MODEL_H_
+#define ORBIT_QT_TRACK_TYPE_ITEM_MODEL_H_
+
+#include <QAbstractTableModel>
+#include <QIcon>
+#include <QModelIndex>
+#include <QObject>
+#include <QString>
+#include <QVariant>
+#include <QtCore>
+#include <map>
+#include <vector>
+
+#include "Track.h"
+#include "TrackManager.h"
+
+namespace orbit_qt {
+
+class TrackTypeItemModel : public QAbstractTableModel {
+  Q_OBJECT
+ public:
+  enum class Column { kVisibility, kName, kEnd };
+
+  explicit TrackTypeItemModel(QObject* parent = nullptr);
+
+  [[nodiscard]] int columnCount(const QModelIndex& parent = {}) const override;
+  [[nodiscard]] QVariant data(const QModelIndex& idx, int role = Qt::DisplayRole) const override;
+  [[nodiscard]] bool setData(const QModelIndex& idx, const QVariant& value,
+                             int role = Qt::EditRole) override;
+  [[nodiscard]] QVariant headerData(int section, Qt::Orientation orientation,
+                                    int role = Qt::DisplayRole) const override;
+  [[nodiscard]] int rowCount(const QModelIndex& parent = {}) const override;
+  [[nodiscard]] Qt::ItemFlags flags(const QModelIndex& index) const override;
+
+  void SetTrackManager(TrackManager* track_manager);
+
+ private:
+  TrackManager* track_manager_ = nullptr;
+  std::vector<Track::Type> known_track_types_;
```

Comment:
About putting it into the cpp file (same applies to `GetTrackTypeDisplayName` below): I think this is a design choice, and I tend to stick with keeping it as non-static member of the class, because it is a property of the class instance - it's currently effectively `const` and can't be configured, but that may change in the future, either by exposing it to the API or subclassing the model. IMO your suggestion makes this less extensible because it implies that all `TrackTypeItemModel` instances always show the same track types.

Making it `const` is nevertheless an option as long as it can't be changed, agreed.